src/TriggerBinding/SqlTriggerScaleMonitor.cs

```diff
@@ -0,0 +1,184 @@
+                return status;
+            }
+
+            // Consider only the most recent batch of samples in the rest of the method.
+            metrics = metrics.TakeLast(minSamplesForScaling).ToArray();
+
+            string counts = string.Join(", ", metrics.Select(metric => metric.UnprocessedChangeCount));
+            this._logger.LogDebugWithThreadId($"Unprocessed change counts: [{counts}], worker count: {workerCount}, maximum changes per worker: {this._maxChangesPerWorker}.");
+
+            // Add worker if the count of unprocessed changes per worker exceeds the maximum limit.
+            long lastUnprocessedChangeCount = metrics.Last().UnprocessedChangeCount;
+            if (lastUnprocessedChangeCount > workerCount * this._maxChangesPerWorker)
+            {
+                status.Vote = ScaleVote.ScaleOut;
+                this._logger.LogInformation($"Requesting scale-out: Found too many unprocessed changes: {lastUnprocessedChangeCount} for table: '{this._userTable.FullName}' relative to the number of workers.");
+                return status;
+            }
+
+            // Check if there is a continuous increase or decrease in count of unprocessed changes.
+            bool isIncreasing = true;
+            bool isDecreasing = true;
+            for (int index = 0; index < metrics.Length - 1; index++)
+            {
+                isIncreasing = isIncreasing && metrics[index].UnprocessedChangeCount < metrics[index + 1].UnprocessedChangeCount;
+                isDecreasing = isDecreasing && (metrics[index + 1].UnprocessedChangeCount == 0 || metrics[index].UnprocessedChangeCount > metrics[index + 1].UnprocessedChangeCount);
+            }
+
+            if (isIncreasing)
+            {
+                // Scale out only if the expected count of unprocessed changes would exceed the combined limit after 30 seconds.
+                DateTime referenceTime = metrics[metrics.Length - 1].Timestamp - TimeSpan.FromSeconds(30);
+                SqlTriggerMetrics referenceMetric = metrics.First(metric => metric.Timestamp > referenceTime);
+                long expectedUnprocessedChangeCount = (2 * metrics[metrics.Length - 1].UnprocessedChangeCount) - referenceMetric.UnprocessedChangeCount;
+
+                if (expectedUnprocessedChangeCount > workerCount * this._maxChangesPerWorker)
+                {
+                    status.Vote = ScaleVote.ScaleOut;
+                    this._logger.LogInformation($"Requesting scale-out: Found the unprocessed changes for table: '{this._userTable.FullName}' to be continuously increasing" +
+                        " and may exceed the maximum limit set for the workers.");
+                    return status;
+                }
+                else
+                {
+                    this._logger.LogDebugWithThreadId($"Avoiding scale-out: Found the unprocessed changes: {expectedUnprocessedChangeCount} for table: '{this._userTable.FullName}' to be increasing" +
```

Comment:
Since all the rest are information it seems like this should match. 

But I'm also thinking we may want to make these all debug to avoid a bunch of noisy messages in the logs. What do other extensions do?

Reply:
It might have been Debug level(I just moved this code from Listener class to this while refactoring) since it is not changing the instance counts.

Reply:
Yeah, that makes sense I suppose. Alright, I'll let you do what you think makes sense and then we can keep an eye here once we start actually getting people using it to see if it'd be helpful to change these.